### PR TITLE
Fabo/2050 404 on missing block

### LINKED ISSUE #2050: Error when redirecting to block > currentHeight
**Describe the bug**
_A clear and concise description of what the bug is._

Error `n.block is undefined` when going to a page block > current height

**To Reproduce**
_Steps to reproduce the behavior:_

1. Go to '/#/blocks/X', where X > current height

**Expected behavior**
_A clear and concise description of what you expected to happen._

Display 404 page or proper error handling

**Additional context**
_Add any other context about the problem here._

Ref: [Sentry](https://sentry.io/share/issue/f5dcc5e29de640beb307f1989d4b5d68/)

### PR DESCRIPTION
**Pull request by faboweb**
Closes #2050 

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: app/src/renderer/components/network/PageBlock.vue [hunk 1]
```diff
@@ -94,8 +94,15 @@ export default {
...
   methods: {
-    async getBlock() {
-      await this.$store.dispatch(`queryBlockInfo`, this.$route.params.height)
+    async getBlock({ $store, $route, $router } = this) {
+      const block = await $store.dispatch(
+        `queryBlockInfo`,
+        $route.params.height
+      )
+
+      if (!block) {
```
Comment (jbibla) [inline app/src/renderer/components/network/PageBlock.vue]: should we include something that makes sure the block number is greater than the current block? not receiving a block from this request could be other issues aside from 404, right?

**Comment by jbibla** [comment]
only issue: `Submodule cosmos-sdk added at c8438a`

File: app/src/renderer/components/network/PageBlock.vue [hunk 2]
```diff
@@ -81,21 +81,37 @@ export default {
...
       return `#` + num.prettyInt(block.block.header.height)
     },
     blockTime({ moment, block } = this) {
+      if (!block.block) return `n/a`
       return moment(block.block.header.time).format(`MMM Do YYYY, HH:mm:ss`)
     }
   },
   watch: {
-    $route: `getBlock`
+    "$route.params.height": function() {
```
Comment (fedekunze) [inline app/src/renderer/components/network/PageBlock.vue]: you can watch the `lastHeader` instead here

File: app/src/renderer/components/network/PageBlock.vue [hunk 3]
```diff
@@ -81,21 +81,37 @@ export default {
...
+    async getBlock({ $store, $route, $router, lastHeader } = this) {
+      // query first for the block so we don't fail if the user started from this route and hasn't received any lastHeader yet
+      const blockInfo = await $store.dispatch(
+        `queryBlockInfo`,
+        $route.params.height
+      )
+
+      if (
+        !blockInfo &&
+        Number($route.params.height) > Number(lastHeader.height)
```
Comment (fedekunze) [inline app/src/renderer/components/network/PageBlock.vue]: shouldn't we have this `if` condition before the dispatch ?

File: app/src/renderer/components/network/PageBlock.vue (hunk 2 above)
Comment (faboweb) [reply app/src/renderer/components/network/PageBlock.vue]: No, this is watching if the route changes, not the chain height

File: app/src/renderer/components/network/PageBlock.vue (hunk 3 above)
Comment (faboweb) [reply app/src/renderer/components/network/PageBlock.vue]: I even commented the reason above ;)
> query first for the block so we don't fail if the user started from this route and hasn't received any lastHeader yet

**Pull request merged**